### PR TITLE
Update imagecodecs requirement to 2023.7.10

### LINKED ISSUE #115: tifffile 2023.8.12 needs imagecodecs>=2023.7.10 to read tiff with deflate
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

In an environment with:
```
imagecodecs               2023.1.23        py39h2c87f5b_0    conda-forge
tifffile                  2023.8.12          pyhd8ed1ab_0    conda-forge
```
`tifffile.imread()` fails to read a TIFF with deflate compression

```
ValueError: <COMPRESSION.ADOBE_DEFLATE: 8> requires the 'imagecodecs' package
```

Simple example
```python
import tifffile
# wget https://github.com/mantidproject/mantidimaging-data/raw/128x128/ISIS/IMAT/IMAT00010675/Tomo/IMAT_Flower_Tomo_000000.tif
data = tifffile.imread("IMAT_Flower_Tomo_000000.tif")
print(data.shape, data.dtype)
```
In the `tifffile` repo https://github.com/cgohlke/tifffile/blob/master/setup.py it requires `imagecodecs>=2023.8.12`. But this is not on conda-forge yet. However `imagecodecs=2023.7.10` does work for me.

Good:
`tifffile=2023.7.18 imagecodecs=2023.1.23`
`tifffile=2023.8.12 imagecodecs=2023.7.10`

Bad
`tifffile=2023.8.12 imagecodecs= 2023.1.23`

### Installed packages

```shell
# packages in environment at /home/sam/mambaforge/envs/test2:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
aom                       3.5.0                h27087fc_0    conda-forge
blosc                     1.21.4               h0f2a231_0    conda-forge
brotli                    1.0.9                h166bdaf_9    conda-forge
brotli-bin                1.0.9                h166bdaf_9    conda-forge
brunsli                   0.1                  h9c3ff4c_0    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.19.1               hd590300_0    conda-forge
c-blosc2                  2.10.0               hb4ffafa_0    conda-forge
ca-certificates           2023.7.22            hbcca054_0    conda-forge
cfitsio                   4.2.0                hd9d235c_0    conda-forge
charls                    2.4.2                h59595ed_0    conda-forge
dav1d                     1.2.1                hd590300_0    conda-forge
giflib                    5.2.1                h0b41bf4_3    conda-forge
imagecodecs               2023.1.23       py310h241fb82_2    conda-forge
jxrlib                    1.1                  h7f98852_2    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
lcms2                     2.15                 haa2dc70_1    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
lerc                      4.0.0                h27087fc_0    conda-forge
libaec                    1.0.6                hcb278e6_1    conda-forge
libavif                   0.11.1               h8182462_2    conda-forge
libblas                   3.9.0           17_linux64_openblas    conda-forge
libbrotlicommon           1.0.9                h166bdaf_9    conda-forge
libbrotlidec              1.0.9                h166bdaf_9    conda-forge
libbrotlienc              1.0.9                h166bdaf_9    conda-forge
libcblas                  3.9.0           17_linux64_openblas    conda-forge
libcurl                   8.2.1                hca28451_0    conda-forge
libdeflate                1.18                 h0b41bf4_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.1.0               he5830b7_0    conda-forge
libgfortran-ng            13.1.0               h69a702a_0    conda-forge
libgfortran5              13.1.0               h15d22d2_0    conda-forge
libgomp                   13.1.0               he5830b7_0    conda-forge
libjpeg-turbo             2.1.5.1              h0b41bf4_0    conda-forge
liblapack                 3.9.0           17_linux64_openblas    conda-forge
libnghttp2                1.52.0               h61bc06f_0    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libopenblas               0.3.23          pthreads_h80387f5_0    conda-forge
libpng                    1.6.39               h753d276_0    conda-forge
libsqlite                 3.42.0               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.1.0               hfd8a6a1_0    conda-forge
libtiff                   4.5.1                h8b53f26_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libwebp-base              1.3.1                hd590300_0    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
libzopfli                 1.0.3                h9c3ff4c_0    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
ncurses                   6.4                  hcb278e6_0    conda-forge
numpy                     1.25.2          py310ha4c1d20_0    conda-forge
openjpeg                  2.5.0                hfec8fc6_2    conda-forge
openssl                   3.1.2                hd590300_0    conda-forge
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
python                    3.10.12         hd12c33a_0_cpython    conda-forge
python_abi                3.10                    3_cp310    conda-forge
readline                  8.2                  h8228510_1    conda-forge
setuptools                68.0.0             pyhd8ed1ab_0    conda-forge
snappy                    1.1.10               h9fff704_0    conda-forge
tifffile                  2023.8.12          pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
wheel                     0.41.1             pyhd8ed1ab_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
zfp                       1.0.0                h27087fc_3    conda-forge
zlib-ng                   2.0.7                h0b41bf4_0    conda-forge
```

### Environment info

```shell
active environment : test2
    active env location : /home/sam/mambaforge/envs/test2
            shell level : 3
       user config file : /home/sam/.condarc
 populated config files : /home/sam/mambaforge/.condarc
                          /home/sam/.condarc
          conda version : 23.3.1
    conda-build version : not installed
         python version : 3.10.6.final.0
       virtual packages : __archspec=1=x86_64
                          __cuda=12.0=0
                          __glibc=2.35=0
                          __linux=6.2.0=0
                          __unix=0=0
       base environment : /home/sam/mambaforge  (writable)
      conda av data dir : /home/sam/mambaforge/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /home/sam/mambaforge/pkgs
                          /home/sam/.conda/pkgs
       envs directories : /home/sam/mambaforge/envs
                          /home/sam/.conda/envs
               platform : linux-64
             user-agent : conda/23.3.1 requests/2.28.1 CPython/3.10.6 Linux/6.2.0-26-generic ubuntu/22.04.3 glibc/2.35
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by samtygier-stfc**
Fixes #115

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by samtygier-stfc** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,7 +26,7 @@ requirements:
   run:
     - python >=3.9
     - numpy >=1.19.2
-    - imagecodecs >=2023.1.23
+    - imagecodecs >=2023.7.10
```
Comment (ericpre) [inline recipe/meta.yaml:29]:
```suggestion
    - imagecodecs >=2023.8.12
```
According to https://github.com/cgohlke/tifffile/commit/19dfae260223c8d190fc483c7e8e6a6fcb2e0151? However, this version is not available yet: https://github.com/conda-forge/imagecodecs-feedstock/pull/86.

Comment (samtygier-stfc) [reply recipe/meta.yaml:29]: imagecodecs 2023.8.12 is available now. so lets go with `imagecodecs >=2023.8.12`

**Comment by ericpre** [comment]
Thank you @samtygier-stfc!

**Pull request merged**